dev-setup/test_bd_close_gated.sh

```
@@ -0,0 +1,93 @@
+}
+
+# ---------------------------------------------------------------------------
+echo "[1] Happy path — unblocked bead closes cleanly"
+solo_id="$(bd create "solo bead" -t task --json | jq -r '.id')"
+set +e
+"$WRAPPER" "$solo_id" --reason "test solo close" >/dev/null
+rc=$?
+set -e
+check "solo close returns 0" 0 "$rc"
+status="$(bd show "$solo_id" --json | jq -r '.[0].status')"
+[[ "$status" == "closed" ]] && echo "  PASS: solo bead marked closed" || { echo "  FAIL: solo bead status=$status"; fail=$((fail + 1)); }
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)